RAG/chroma_embedding.py

```diff
@@ -166,13 +163,22 @@ def get_similar_documents(self, query) -> list[tuple[float, int, str]]:
         Returns:
             object: Retrieved documents from the Chroma database.
         """
-        # TODO restrict return doc amount to self.__num_matches
-        docs = self.__chroma_db.similarity_search(query)
-        parsed_docs = []
-        for doc in tqdm(docs, desc="Parsing retrieved documents"):  # Add progress bar
-            parsed_docs.append(
-                (0.0, doc.metadata['seq_num'], doc.page_content))
-        return parsed_docs
+        if rerank:
+            print("Reranking documents...")
+            return self.__rerank(query_text, top_n=top_n,
+                                 search_kwargs=search_kwargs)
+
+        return self.__chroma_db.similarity_search(query_text)
+
+    def __rerank(self, prompt, top_n=3, search_kwargs=20):
```

Comment:
Why top 3? Shouldn't we do more

Reply:
Cause of context window right. I would rather have top 3. U have seen the needle in the haystack metric right

Reply:
Can you elaborate? Also if we switch to 4o which has a longer context window, we should def pass in more context. 

The reason I think more context is better in this case is that we aren't passing in few shot examples, instead we are trying to pass in more medical info that can be useful for the rag response